### DISCUSSION TITLE
Scripting Help needed

### DISCUSSION THREAD

**Discussion by mmmmm009** [arendst/Tasmota#20812] [Q&A] [answered]
Hi,
How can I modify the following script to do part between ???   ??? only when Volt has changed?
Thx.

`
...

>F
val=adc(2)
Volt=248+(5*val/1000);

???
if Volt>250
then
pwm1=(Volt-250)*100;
if pwm1>100
then
pwm1=100
endif
endif
=>dimmer %pwm1%
???

`

**Comment by gemu2015** [comment, marked as answer]
```
if chg[Volt]>0
then
if Volt>250
then
pwm1=(Volt-250)*100;
if pwm1>100
then
pwm1=100
endif
endif
endif
```

**Reply by mmmmm009** [reply]
thx!
